hack/unwanted-dependencies.json

```diff
@@ -75,7 +75,6 @@
   "status": {
     "unwantedReferences": {
       "cloud.google.com/go": [
-        "cloud.google.com/go/compute",
```

Comment:
huh... I'm happy to see this drop out, but later / current versions of `cloud.google.com/go/compute` still depend on `cloud.google.com/go`, so I won't be surprised to see it come back.

The big win will be when `cloud.google.com/go/compute/metadata` stops depending on `cloud.google.com/go/compute`, and all the other things we have depending on `cloud.google.com/go/compute` below start dropping out. I think that will happen early next year along with the genproto cleanup.

Reply:
Uff sad to hear that, the problem is that with that inside the test fails and this update is required to fix the security issue
@liggitt how do you think we can proceed with this?

Reply:
oh, I wouldn't block on this, dropping this here and then adding it back later if we need to is fine

@dims said there's another CVE that needs an even newer grpc - https://github.com/kubernetes/kubernetes/pull/121338 was open for that ... should we take that one?

Reply:
I started with the even newer version 1.59.0 but that was a lot of more changes required including s2a-go package which it's also unwanted, if we can go with 1.59.0 and remove that unwanted one it's ok
The version 1.56.3 it's the one that requires less changes in the code and manage to fix the security issue with few changes

Reply:
the s2a-go issue in particular was discussed in slack [here](https://kubernetes.slack.com/archives/CHGFYJVAN/p1696544918092889):

> John Howard
> The new version of [cloud.google.com/go/compute](http://cloud.google.com/go/compute) pulls in an "unwanted dependency" s2a-go. Any action we need to do here? Remove from unwanted? Complain to go/compute repo? Don't update go/compute? Running into this in https://github.com/kubernetes/kubernetes/pull/120241. Can work around it in that PR sort of, though.
> 
> Stephen Kitt
> ... or add it to the status for [cloud.google.com/go/compute](http://cloud.google.com/go/compute) in unwanted-dependencies.json — the former is already an unwanted module, so adding a cloud dependency to an existing unwanted cloud dependency doesn’t change much in the grand scheme of things IMO
> 
> Stephen Kitt
What would be really unwanted would be adding s2a-go as a dependency for a non-unwanted module, or a direct k/k dependency.
> 
> liggitt
> there were two reasons it was marked as unwanted:
> * unstable (due to vestigial content in the readme from when it was under development)  that has since been removed - https://github.com/google/s2a-go/pull/120
> * a transitive dependency it had on appengine code (cloud dependencies) that has since been removed - https://github.com/google/s2a-go/pull/119
> 
> the current state of that library means it doesn't pull in objectionable transitive dependencies any more
that said, we don't want more dependencies in general, so we look hard at any additions. in this case, it looks like the only referencer is [cloud.google.com/go/compute](http://cloud.google.com/go/compute), which we have a timeline of ~next March for dropping. +1 to Stephen Kitt’s comment about tracking the things that pull in s2a-go and limiting it to cloud deps we have a plan to drop, so this will drop out along with them

Reply:
I wasn't in that channel, ok so it make sense to restrict the s2a-go dependency to the already existed dependency that as I checked it was only the go/compute thing, probably @dims can target version 1.59.0 on his PR and we can drop this one that actually fix only the gprc because I thought #121338 was only about OTEL :D 
Closing this one to let the other one proceed